sub-packages/bionemo-example_model/src/bionemo/example_model/training_scripts/finetune_mnist.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+from pathlib import Path
+
+from nemo import lightning as nl
+from nemo.collections import llm
+from nemo.lightning import NeMoLogger, resume
+from nemo.lightning.pytorch import callbacks as nl_callbacks
+from pytorch_lightning.loggers import CSVLogger, TensorBoardLogger
+
+from bionemo.example_model.lightning.lightning_basic import (
+    BionemoLightningModule,
+    ExampleFineTuneConfig,
+    data_module,
+)
+
+
+def run_finetune(checkpoint_dir: str, name: str, directory_name: str):
+    """Run the finetuning step.
+
+    Args:
+        checkpoint_dir: The directory with the previous model
+        name: The experiment name.
+        directory_name: The directory to write the output
+    Returns:
+        str: the path of the trained model.
+    """
+    save_dir = Path(directory_name) / "classifier"
+    checkpoint_callback = nl_callbacks.ModelCheckpoint(
+        save_last=True,
+        save_on_train_epoch_end=True,
+        monitor="reduced_train_loss",
+        every_n_train_steps=25,
+        always_save_context=True,  # Enables the .nemo file-like checkpointing where all IOMixins are under SerDe
+    )
+
+    nemo_logger2 = NeMoLogger(
+        log_dir=str(save_dir),
+        name=name,
+        tensorboard=TensorBoardLogger(save_dir=save_dir, name=name),
+        ckpt=checkpoint_callback,
+        extra_loggers=[CSVLogger(save_dir / "logs", name=name)],
+    )
+
+    lightning_module2 = BionemoLightningModule(
+        config=ExampleFineTuneConfig(
+            initial_ckpt_path=checkpoint_dir,
+            initial_ckpt_skip_keys_with_these_prefixes={"digit_classifier"},
+        )
+    )
+
+    strategy = nl.MegatronStrategy(
```

Comment:
For my own knowledge, what's the difference between nemo lightning (`nl`) and pytorch lightning `pl`?

Reply:
A lot of nemo lightning seems to extend pytorch lightning, often with Megatron.